HTML5SDK/wwtlib/Layers/FitsImage.cs

```diff
@@ -116,10 +117,25 @@ public static bool IsGzip(BinaryReader br)
             }
         }
 
+        private bool ValidateFitsSimple (BinaryReader br)
+        {
+            int pos = br.Position;
+            br.Seek(0);
+            string data = br.ReadByteString(8);
+            string keyword = data.TrimEnd();
+            br.Seek(pos);
+            return keyword.ToUpperCase() == "SIMPLE";
+        }
+
         public void ParseHeader(BinaryReader br)
         {
-            bool foundEnd = false;
+            if (!ValidateFitsSimple(br))
+            {
+                Script.Literal("alert('The requested file is not a valid FITS file.')");
```

Comment:
Can you change this from alert to console.warn and throw an error? That way you can catch the error in your script implementation and alert it on your end and the web-client can show the error as highlighted text in a modal dialog.

Reply:
@thewebkid The problem with using throw/catch for errors here is that because the file loading and parsing is launched asynchronously in the backend, any errors thrown during the file parsing would have to be caught and dealt with in the backend. We should have a better way to notify users of errors than using alerts in the backend but using errors will not work with the current code.

Reply:
Makes sense to me.